### PR TITLE
Traps + Code Cleaning Duty

### PR DESCRIPTION
**Pull request by V2LenKagamine**
## About The Pull Request
Technical Changes:
 - Removes old define regarding ERP
 - New Trait 'Light Step' for dungeon traps, given to rogues.
 - Cleans up `/obj/structure/bonepile` and gives a `!` icon during mapping
 - - Also adds inert variant for pretties.
 - Adds some nullchecks to ammunition to support `null` users.
 - - Adds nullcheck to wizard fireball too, for reasons.
 - Removes Duplicate Demon Definition in `underworld.dm`
 
Completely New Traps - Replacing the old ones [They weren't used in any merged maps/areas.]
Traps have 2 trigger types: Step On/Step Off.
 Will not trigger for those with "Light Step"
 - Frost trap: Freezes the victim, causing them to take 2x damage if hit before the stun wears off!
 - Fire trap: "I appear to have burst into flames."
 - Poison trap: Poisons victim with 2.5u of berry poison
 - Spike trap : Just deals flat brute and stuns.
 - Wall_projectile trap: shoots a projectile from a wall based on what direction it's placed!
 - - Default a stone arrow
 - - Comes with mapping-helper fireball!
 - Bomb trap: Goes click, until you step off, then boom.
 - Saw trap: Goes click, until you step off, then saw blade.

 - Global Comsig for 'Active Perception' that traps listen to. This might be bad-code.
## Why It's Good For The Game

Main Man request. Also Funny traps for mapping.

## Pre-Merge Checklist
- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.

### CODE REVIEW

File: code/game/objects/structures/traps.dm [hunk 1]
```diff
@@ -39,165 +48,172 @@
...
+	if(charges == 0)
+		animate(src, alpha = 0, time = 5)
+		QDEL_IN(src, 5)
 	return
+///Common checks to make sure we can trigger the trap.
+/// True == Yep we good.
+/obj/structure/trap/proc/trap_check(mob/living/victim)
+	if(last_trigger + time_between_triggers > world.time)
+		return FALSE
+	if(victim.mind in immune_minds || HAS_TRAIT(victim,TRAIT_LIGHT_STEP))
```
Comment (dwasint) [inline code/game/objects/structures/traps.dm:104]: put the in statement in its own ()

File: code/game/objects/structures/traps.dm [hunk 2]
```diff
@@ -39,165 +48,172 @@
...
-	to_chat(user, "<span class=notice>I set up [src]. Examine while close to disarm it.</span>")
-	stored_trap.forceMove(T)//moves trap to ground
-	forceMove(stored_trap)//moves item into trap
-
-/obj/item/bountytrap/Destroy()
-	qdel(stored_trap)
-	QDEL_NULL(spark_system)
-	. = ..()
+	victim.electrocute_act(30, src, flags = SHOCK_NOGLOVES) // electrocute act does a message.
+	victim.Paralyze(5 SECONDS)
```
Comment (dwasint) [inline code/game/objects/structures/traps.dm:133]: put the actual stun as a define so mappers can modify this if they want

**Review by dwasint — CHANGES_REQUESTED** [review]
also stop putting unrelated fixes in content prs

File: code/game/objects/structures/traps.dm (hunk 1 above)
Comment (V2LenKagamine) [reply code/game/objects/structures/traps.dm:104]: Will do

File: code/game/objects/structures/traps.dm (hunk 2 above)
Comment (V2LenKagamine) [reply code/game/objects/structures/traps.dm:133]: Adding Define

**Comment by V2LenKagamine** [comment]
> also stop putting unrelated fixes in content prs

No :smiel:
[Sorry,I hate small 1 line fix Pr's so i'll often try to bundle in fixes that are semi-related, like how the projectile system didnt support null users was originally so the traps could fire them before I moved over to just spawning in projectiles manually; but it ended up still being needed for the redstone activator.]

**Pull request merged**